File: admin-ui/src/services/swagger/dataScopeApi.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request';

/** 获取数据权限 GET /api/data-scopes/${param0}/${param1} */
export async function getDataScopes(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getDataScopesParams,
  options?: { [key: string]: any },
) {
  const { targetType: param0, targetId: param1, ...queryParams } = params;
  return request<API.DataScopeDTO[]>(`/api/data-scopes/${param0}/${param1}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 数据赋权 PUT /api/data-scopes/grant */
export async function grantDataScope(
  body: API.GrantDataScopeRequestDTO,
  options?: { [key: string]: any },
) {
  return request<any>('/api/data-scopes/grant', {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 获取用户数据权限 GET /api/data-scopes/users/${param0} */
export async function getUserDataScopes(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getUserDataScopesParams,
  options?: { [key: string]: any },
) {
  const { userId: param0, ...queryParams } = params;
  return request<API.DataScopeDTO[]>(`/api/data-scopes/users/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

File: admin-backend/application-api/src/main/java/com/xfrog/platform/application/permission/api/DataScopeApi.java
package com.xfrog.platform.application.permission.api;

import com.xfrog.platform.application.permission.api.dto.DataScopeDTO;
import com.xfrog.platform.application.permission.api.dto.GrantDataScopeRequestDTO;
import com.xfrog.platform.domain.share.permission.DataScopeTargetType;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import java.util.List;

@Tag(name = "DataScopeApi", description = "数据权限接口")
@RequestMapping("/api/data-scopes")
public interface DataScopeApi {
    @PutMapping("/grant")
    @Operation(summary = "数据赋权")
    void grantDataScope(@Valid @RequestBody GrantDataScopeRequestDTO requestDTO);

    @GetMapping("/{targetType}/{targetId}")
    @Operation(summary = "获取数据权限")
    List<DataScopeDTO> getDataScopes(@PathVariable("targetType") @Valid @NotNull DataScopeTargetType targetType, @PathVariable("targetId") Long targetId);

    @GetMapping("/users/{userId}")
    @Operation(summary = "获取用户数据权限")
    List<DataScopeDTO> getUserDataScopes(@PathVariable("userId") Long userId);
}
